src/Form/Form.js

```diff
@@ -381,6 +385,8 @@ class Form {
             const input = e.target
             let click = null
 
+            if (!canBeInteractedWith(input)) return
```

Comment:
This whole thing means: instead of decorating/not decorating, let's focus on the click handler (here). If the element is disabled/readonly we return here. This has the benefit that when the disabled/readonly changes the handler just works.